### ISSUE TITLE
Add ingress-sidecar node type.

### ISSUE THREAD

**Issue by PiotrSikora** [istio/istio#3469]
Provide deployment model, where sidecar can be injected into existing ingress giving it access to the service mesh (with mTLS, etc.), but without intercepting traffic from/to the Internet.

See #3059.

cc @costinm @nmittler @ZackButcher

**Comment by nmittler** [comment]
The workaround in https://github.com/istio/istio/issues/3059#issuecomment-365861182 was to simply disable the iptables redirect for inbound traffic, so that the sidecar is only active on the mesh side. [The downside](https://github.com/istio/istio/issues/3059#issuecomment-365794988) is that services within the mesh won't be able to call back to their ingress.  This is probably fine for the vast majority of use cases. Are we comfortable with this being our story for now (e.g. 1.0)?

**Comment by PiotrSikora** [comment]
@nmittler can you disable it from Istio config or only manually from the pod (which presumably doesn't persist across restarts)?

**Comment by nmittler** [comment]
@PiotrSikora IIUC, the isto-proxy container does not run as privileged, so it can't change iptables. The iptables are currently configured by the `proxy-init` init container. AFAICT there wouldn't be an easy way today of changing this on the fly (but others please chime in if that's incorrect).

**Comment by munnerz** [comment]
Yep I can attest, in order to apply the hack you'll need to change the sidecar (or some other container with iptables installed) to include the NET_ADMIN capability.

FWIW, I've not been able to get nginx-ingress to correctly return a response. As @nmittler commented here (https://github.com/istio/istio/issues/3059#issuecomment-365439442) only a 404 is returned from nginx (with istio-proxy logging NR on one of those requests):

```
[2018-02-13T20:11:26.679Z] "GET /productpage HTTP/1.1" 404 NR 0 0 1 - "127.0.0.1" "curl/7.58.0" "67268b90-3df2-9672-ac9f-12161d4deaf2" "104.154.191.248" "-"
[2018-02-13T20:11:26.654Z] "GET /productpage HTTP/1.1" 404 - 0 0 32 2 "-" "curl/7.58.0" "67268b90-3df2-9672-ac9f-12161d4deaf2" "104.154.191.248" "127.0.0.1:80"
```

I **am** however able to make that same curl request to my productpage backend just fine. This is when **not** using istio-ingress at all (instead I am disabling mTLS auth on the nginx-ingress service, but still injecting the sidecar into the nginx-ingress pod so it is able to route to services within the mesh). I'm aware this would mask the client IP for now, but that is acceptable.

Does this issue also cover fixing the above ^? Is this expected/can anyone shed any light on a workaround? The aforementioned iptables hack above did not resolve my problem when I tried.

**Comment by linsun** [comment]
@nmittler FYI, here is the POC I did on pilot agent config, you may find useful:  https://github.com/linsun/pilot-config

I'd be happy to help/review what you have.  I'm very interested in this feature too.

**Comment by nmittler** [comment]
Great, thanks @linsun!

**Comment by nmittler** [comment]
After some discussion with @ayj, I think the way to solve this is by simply adding some annotations to afford the ability to customize the iptables configuration for a given deployment.
